_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+This is the sweet spot of AWS Elastic Container Service (ECS) Fargate. Fargate gives you networking abstractions across a virtual network known as a VPC (Virtual Private Cloud). This network abstraction is built right into the heart of AWS and is well vetted for any type of workload, including high-security government workloads. Fargate takes this a step further by abstracting away the machine management. You can set up traditional clusters and manage your machines if you want, but leveraging Fargate simplifies one more part of your process. Ultimately our goal with using cloud vendors in the first place is to let them be good at infrastructure management so we can be good at managing our business.
+
+Time to jump into it and try it out using .NET containers with AWS Fargate!
+
+**What you will need to get started**
+
+What you'll need to continue are the following:
+
+- Basic knowledge of .NET
+- [.NET SDK 3.1](https://docs.microsoft.com/en-in/dotnet/core/install/)
+- [An AWS account (we'll be using a free tier product)](https://aws.amazon.com/free)
+- [Okta CLI](https://cli.okta.com)
+- [AWS CLI V2](https://aws.amazon.com/cli/)
+- [Docker Desktop for Windows/macOS (Not required if you are on Linux)](https://www.docker.com/get-started)
+
+This tutorial assumes you already have Docker set up and running.
+
+## Build and Dockerize a .NET Chat Application
+
+So what do you want to build? To keep things simple so we can see the value in using something other than standard HTTP protocols, I'll use SignalR to build a very basic chat application.
+
+- Secure: Only logged-in clients should be able to use the chat functionality
+- Chat users names must come from their validated identity
+- Real-time
+
+To achieve this, I am going to use four technologies:
+
+- Okta for identity management
+- .NET to host the application
+- SignalR to provide the socket management abstraction
+- Vue.js to provide the rendering for the front-end
+
+### Authentication for Your .NET Chat App
+
+Authentication is vital in any application, but doubly so when you need to depend on who someone is. I used to roll all my own security services, but I don't anymore because there are too many threat verticals and too much for me to care to manage. I'd rather delegate that responsibility to another company that can focus solely on those concerns so I can focus entirely on my business. I personally like using Okta for this purpose. I work with customers with many applications across many languages and vendors; Okta makes it really easy to incorporate all of them into one management pipeline to delegate access to those who need it and shut down access across the entire suite of applications if a bad actor gets a set of credentials.
```

Comment:
```suggestion
Authentication is vital in any application, but doubly so when you need to depend on who someone is. I used to roll all my own security services, but I don't anymore because there are too many threat verticals and too much to manage. I'd rather delegate that responsibility to another company that can focus solely on those concerns, so I can focus entirely on my business. I use Okta for this purpose. I work with customers with many applications across many languages and vendors; Okta makes it easy to incorporate all of them into one management pipeline to delegate access to those who need it and shut down access across the entire suite of applications if a bad actor gets a set of credentials.
```

Reply:
i'll rewrite this and yes you are right this comes from previous post